### PR TITLE
Multiarch-CICD support using drone1.0 and docker manifest

### PR DESCRIPTION
**Pull request by jianghang8421**
This PR includes the changes for multiarch support using drone 1.0 and docker manifest.

Issue
https://github.com/rancher/rancher/issues/16461
https://github.com/rancher/rancher/issues/16518

### CODE REVIEW

File: .drone-multiarch.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
+kind: pipeline
+name: amd64
+
+platform:
+  os: linux
+  arch: amd64
+
+steps:
+  - name: build
+    image: rancher/dapper:1.11.2
```
Comment (ibuildthecloud) [inline .drone-multiarch.yml]: switch to dapper:v0.4.1 which is multiarch now

File: .drone-multiarch.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+name: arm64
+
+platform:
+  os: linux
+  arch: arm64
+
+steps:
+  - name: build
+    image: jianghang8421/dapper-arm64:18.06-git
+    volumes:
```
Comment (loganhz) [inline .drone-multiarch.yml]: Please use dapper in our rancher dockerhub repo

File: package/Dockerfile [hunk 3]
```diff
@@ -1,19 +1,33 @@
...
-    && curl -sLf https://github.com/kelseyhightower/confd/releases/download/v0.16.0/confd-0.16.0-linux-amd64 > /usr/bin/confd \
+    && curl -sLf https://github.com/kelseyhightower/confd/releases/download/v0.16.0/confd-0.16.0-linux-${ARCH} > /usr/bin/confd \
     && chmod +x /usr/bin/confd \
-    && curl -sLf https://get.docker.com/builds/Linux/x86_64/docker-1.12.3.tgz | tar xvzf - -C /opt/rke-tools/bin --strip-components=1 docker/docker \
+    && curl -sLf ${!DOCKER_URL} | tar xvzf - -C /opt/rke-tools/bin --strip-components=1 docker/docker \
     && chmod +x /opt/rke-tools/bin/docker \
     && apk del curl
 
+ENV PORTMAP_URL_amd64=https://github.com/projectcalico/cni-plugin/releases/download/v1.9.1/portmap \
+    PORTMAP_URL_arm64=https://github.com/jianghang8421/plugins/releases/download/v1.9.1-rancher/portmap-arm64 \
```
Comment (ibuildthecloud) [inline package/Dockerfile]: what are we going to do about this?

File: .drone-multiarch.yml (hunk 1 above)
Comment (jianghang8421) [reply .drone-multiarch.yml]: Updated to dapper:v0.4.1

File: package/Dockerfile (hunk 3 above)
Comment (jianghang8421) [reply package/Dockerfile]: According to the [release](https://github.com/projectcalico/cni-plugin/releases/tag/v1.9.1 ) which the amd64 arch is using. The portmap plugin is built from [this commit](https://github.com/containernetworking/plugins/commit/e8bea554c5ec9a433d4353c03f7f31b068185c1c), not an official release, so I forked the repo `containernetworking/plugins` and built the arm64 portmap from this commit.
I don't know if we can upgrade the portmap to an official release version which support arm64 arch.

**Pull request merged**